include/deal.II/grid/grid_tools.h

```diff
@@ -750,6 +753,45 @@ namespace GridTools
       &cell_hint =
         typename Triangulation<dim, spacedim>::active_cell_iterator());
 
+  /**
+   * This function is similar to GridTools::compute_point_locations(),
+   * but it tries to find and transform every point of @p points.
+   *
+   * @return A tuple containing four elements; the first three
+   * are documented in GridTools::compute_point_locations().
+   * The last element of the @p return_type contains the
+   * indices of points which are not not found inside the mesh
+   * or lie on artificial cells.
+   *
+   * @code
+   *   std::tuple<
+   *     std::vector<
+   *        typename Triangulation<dim,spacedim>::active_cell_iterator>,
+   *        std::vector<std::vector<Point<dim>>>,
+   *        std::vector<std::vector<unsigned int>>,
+   *        std::vector<unsigned int> >
```

Comment:
This is better, but the indentation of the second, third, and fourth type are now wrong.